src/plugin/indexer-elastic-rest/src/java/org/apache/nutch/indexwriter/elasticrest/ElasticRestIndexWriter.java

```diff
@@ -0,0 +1,336 @@
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.nio.conn.SchemeIOSessionStrategy;
+import org.apache.http.nio.conn.ssl.SSLIOSessionStrategy;
+import org.apache.http.ssl.SSLContextBuilder;
+import org.apache.http.ssl.TrustStrategy;
+import org.apache.nutch.indexer.IndexWriter;
+import org.apache.nutch.indexer.NutchDocument;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLContext;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.net.URL;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+
+/**
+ */
+public class ElasticRestIndexWriter implements IndexWriter {
+    public static Logger LOG = LoggerFactory.getLogger(ElasticRestIndexWriter.class);
+
+    private static final int DEFAULT_MAX_BULK_DOCS = 250;
+    private static final int DEFAULT_MAX_BULK_LENGTH = 2500500;
+
+    private JestClient client;
+    private String defaultIndex;
+    private String defaultType = null;
+
+    private Configuration config;
+
+    private Bulk.Builder bulkBuilder;
+    private Future<HttpResponse> execute;
+    private int port = -1;
+    private String host = null;
+    private String user = null;
+    private Boolean https = null;
+    private String password = null;
+    private Boolean trustAllHostnames = null;
+
+    private int maxBulkDocs;
+    private int maxBulkLength;
+    private long indexedDocs = 0;
+    private int bulkDocs = 0;
+    private int bulkLength = 0;
+    private boolean createNewBulk = false;
+    private long millis;
+    private BasicFuture<JestResult> basicFuture = null;
+
+    @Override
+    public void open(JobConf job, String name) throws IOException {
+
+        host = job.get(ElasticRestConstants.HOST);
+        port = job.getInt(ElasticRestConstants.PORT, 9200);
+        user = job.get(ElasticRestConstants.USER);
+        password = job.get(ElasticRestConstants.PASSWORD);
+        https = job.getBoolean(ElasticRestConstants.HTTPS, false);
+        trustAllHostnames = job.getBoolean(ElasticRestConstants.HOSTNAME_TRUST, false);
+
+        // trust ALL certificates
+        SSLContext sslContext = null;
+        try {
+            sslContext = new SSLContextBuilder().loadTrustMaterial(new TrustStrategy() {
+                public boolean isTrusted(X509Certificate[] arg0, String arg1) throws CertificateException {
+                    return true;
+                }
+            }).build();
+        } catch (NoSuchAlgorithmException | KeyManagementException | KeyStoreException e) {
+            e.printStackTrace();
```

Comment:
Can you actually LOG the stacktrace instead of printing it?
LOG.error("Error detected whilst... blah blah blah", e).
Thank you